Leanplum/Leanplum.h

```diff
@@ -14,6 +14,33 @@ FOUNDATION_EXPORT double LeanplumVersionNumber;
 //! Project version string for Leanplum.
 FOUNDATION_EXPORT const unsigned char LeanplumVersionString[];
 
-// In this header, you should import all the public headers of your framework using statements like #import <Leanplum/PublicHeader.h>
 
+@interface Leanplum : NSObject
 
+/**
+ * Sets additional user attributes after the session has started.
+ * Variables retrieved by start won't be targeted based on these attributes, but
+ * they will count for the current session for reporting purposes.
+ * Only those attributes given in the dictionary will be updated. All other
+ * attributes will be preserved.
+ */
++ (void)setUserAttributes:(NSDictionary *)attributes
+              withSuccess:(void (^)(void))success
+              withFailure:(void (^)(NSError *error))failure;
+
+/**
+ * Updates a user ID after session start.
+ */
++ (void)setUserId:(NSString *)userId
+      withSuccess:(void (^)(void))success
+      withFailure:(void (^)(NSError *error))failure;
+
+/**
+ * Updates a user ID after session start with a dictionary of user attributes.
+ */
++ (void)setUserId:(NSString *)userId withUserAttributes:(NSDictionary *)attributes
+      withSuccess:(void (^)(void))success
+      withFailure:(void (^)(NSError *error))failure;
+
```

Comment:
I feel like the naming here is kind of strange. userId is a part of user attributes, why do we have to have a separate function just for setUserId? Can we have a setUserAttributes(userId) function with just user Id as param? We can have 3 setUserAttributes with varying params here then. But if we have to keep the same methods as the old SDK then this is fine

Reply:
Yes, the goal here is the keep the existing api interfaces as same as possible.